hw/top_earlgrey/data/autogen/top_earlgrey.gen.hjson

```diff
@@ -711,40 +711,28 @@
           inst_type: eflash
           base_addr: 0x20000000
           size_byte: 0x80000
+          xbar: false
           pipeline_byp: "true"
         }
         {
-          name: uart
+          name: peri
           type: device
           clock: clk_fixed_i
           reset: rst_fixed_ni
           pipeline_byp: "false"
-          inst_type: uart
-          base_addr: 0x40000000
-          size_byte: 0x1000
-          pipeline: "true"
-        }
-        {
-          name: gpio
-          type: device
-          clock: clk_fixed_i
-          reset: rst_fixed_ni
-          pipeline_byp: "false"
-          inst_type: gpio
-          base_addr: 0x40010000
-          size_byte: 0x1000
-          pipeline: "true"
-        }
-        {
-          name: spi_device
-          type: device
-          clock: clk_fixed_i
-          reset: rst_fixed_ni
-          pipeline_byp: "false"
-          inst_type: spi_device
-          base_addr: 0x40020000
-          size_byte: 0x1000
+          xbar: true
           pipeline: "true"
+          xbar_addr:
```

Comment:
I will create another PR for mergine `xbar_addr` with `base_addr` and `size_byte` . Meaning regardless of Xbar device port or IP (leaf) device port, the node in Hjson has `addr_range` that is list of addresses (sets of `base_addr` and `size_byte`).